src/providers/signatureHelpProvider.ts

```diff
@@ -0,0 +1,155 @@
+import * as vscode from 'vscode';
+import { HoverRequest, LanguageClient } from 'vscode-languageclient';
+
+export class SignatureHelpProvider implements vscode.SignatureHelpProvider {
+  private languageClient: LanguageClient;
+  private previousFunctionPosition?: vscode.Position;
+
+  constructor(lc: LanguageClient) {
+    this.languageClient = lc;
+  }
+
+  public provideSignatureHelp(
+    document: vscode.TextDocument,
+    position: vscode.Position,
+    token: vscode.CancellationToken,
+    context: vscode.SignatureHelpContext,
+  ): vscode.ProviderResult<vscode.SignatureHelp> {
+    // the current signature help provider uses the hover information from RLS
+    // and it only has a string representation of the function signature.
+    // This check makes sure we can easily show the tooltip for multiple parameters, separated by `,`
+    if (context.triggerCharacter === '(') {
+      this.previousFunctionPosition = position;
+      return this.provideHover(
+        this.languageClient,
+        document,
+        position,
+        token,
+      ).then(hover => {
+        return this.hoverToSignatureHelp(hover);
```

Comment:
For such simple functions I'd prefer to use the `() => returnValue` shorthand. I think we should enable https://palantir.github.io/tslint/rules/arrow-return-shorthand/ rule in some form or another later on but let's not worry about that now :+1:

Reply:
Definitely more readable this way, updating.